### PR TITLE
netopt/drivers: Add phy link status output

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds netopt support to query the status of the phy link of a network driver, whether a cable is attached and the link is functional. Output for the ifconfig command to report this status is added.

### Issues/PRs references

None

### Todo

Other drivers beside the enc28j60:

- [ ] netdev_tap
- [x] encx24j600
~~w5100~~

### CODE REVIEW

**Comment by bergzand** [comment]
The w5100 doesn't seem to support reporting link status information :(

**Comment by miri64** [comment]
Maybe @kaspar030 can test?

**Comment by bergzand** [comment]
Netdev_tap support seems a bit complicated. While it is possible to put the tap device in a `down` state on the OS side, I couldn't retrieve the status from within RIOT.

**Comment by basilfx** [comment]
Tested this with a `enc28j60` with `tests/driver_ enc28j60`.

schermafbeelding 2018-03-16 om 20 52 23

File: sys/include/net/netopt.h [hunk 1]
```diff
@@ -270,6 +270,17 @@ typedef enum {
      */
     NETOPT_IS_WIRED,
 
+    /**
+     * @brief Get the Phy link status.
```
Comment (miri64) [inline sys/include/net/netopt.h]: Can you adapt for the documentation style in #8655, please?

Comment (bergzand) [reply sys/include/net/netopt.h]: Of course, I've adapted the `@brief` to the style of #8655 and reworded the full description a bit to specify behaviour more precise.

**Comment by miri64** [comment]
(If `netdev_tap` is somehow possible, we always can do this in a follow-up)

**Comment by bergzand** [comment]
Squashed!

> (If netdev_tap is somehow possible, we always can do this in a follow-up)

Yes, I probably just lack the correct knowledge about how to get the information via some syscall.

**Pull request merged**